server/polar/models/event_metrics.py

```diff
@@ -0,0 +1,205 @@
+"""
+SQLAlchemy model for the events_metrics_hourly continuous aggregate.
+
+This is a read-only materialized view that pre-computes event metrics
+at hourly granularity. With real-time aggregation enabled, queries
+automatically include fresh data from the raw hypertable for the
+most recent un-materialized period.
+"""
+
+import datetime
+from collections.abc import Generator
+from decimal import Decimal
+from uuid import UUID
+
+from alembic_utils.replaceable_entity import ReplaceableEntity, register_entities
+from sqlalchemy import TIMESTAMP, Index, Numeric, String, Uuid, text
+from sqlalchemy.orm import Mapped, Session, mapped_column
+from sqlalchemy.sql.elements import TextClause
+
+from polar.kit.db.models import Model
+
+
+# https://github.com/candidhealth/alembic-utils-extended/blob/9b48280a221e9861aa2f4d0751b8f73ddc5ca005/src/alembic_utils_extended/pg_materialized_view.py
+# TimescaleDBContinuousAggregate is licensed under the MIT License
+#
+# Copyright (c) 2020 Candid Health
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+class TimescaleDBContinuousAggregate(ReplaceableEntity):
```

Comment:
This is a bit of a hodgepodge of things. I will understand if someone comes for me at night